smoke-test/action.yml

```diff
@@ -39,12 +42,59 @@ inputs:
 runs:
   using: "composite"
   steps:
+  - name: Extract Chart YAML
+    id: extract-chart-yaml
+    if: inputs.version == ''
+    shell: bash
+    run: |
+      tarball=$(find ${YAML_DIR} -name "${CHART_NAME}*.tgz")
+      chart_yaml=$(tar -xzvf ${tarball} $(tar -tzf ${tarball} --exclude "*/charts/*" | grep Chart.yaml))
+      echo "CHART_YAML=${chart_yaml}" >> $GITHUB_ENV
+    env:
+      YAML_DIR: ${{ inputs.yaml-dir }}
+      CHART_NAME: ${{ inputs.helm-chart-name }}
+  - name: Get Chart Version
+    id: get-chart-version
+    if: inputs.version == ''
+    uses: mikefarah/yq@master
+    with:
+      cmd: yq '.version' ${CHART_YAML}
   - name: Define App Version
     shell: bash
-    run: echo "APP_VERSION=0.0.1-${GITHUB_REF_NAME//[^a-zA-Z0-9]/}.${GITHUB_RUN_ID}${GITHUB_RUN_ATTEMPT}" >> $GITHUB_ENV
+    if: inputs.version == ''
+    run: echo "APP_VERSION=${CHART_VERSION}+${GITHUB_REF_NAME//[^a-zA-Z0-9]/}.${GITHUB_RUN_ID}${GITHUB_RUN_ATTEMPT}" >> $GITHUB_ENV
+    env:
+      CHART_VERSION: ${{ steps.get-chart-version.outputs.result }}
```

Comment:
Other idea: If `installation-method` is `helm`, extract the version from `Chart.yaml`. Doing that would also remove the extra requirement to specify an input version.

Reply:
It's actually an either/or. I just flubbed making the version required. It will use the one you provide, or extract it and update it to a build specific version number.

Reply:
yes, extended out. the conditional